dds/idl/typeobject_generator.cpp

```diff
@@ -1762,7 +1762,11 @@ typeobject_generator::generate(AST_Type* node, UTL_ScopedName* name)
     gti.endArgs();
     const OpenDDS::XTypes::TypeIdentifier ti = get_minimal_type_identifier(node);
     be_global->impl_ <<
-      "  static const XTypes::TypeIdentifier ti = " << ti << ";\n"
+      "  static XTypes::TypeIdentifier ti;\n"
+      "  ACE_GUARD_RETURN(ACE_Thread_Mutex, guard, TheServiceParticipant->get_tm_lock(), ti);\n"
+      "  if (ti.kind() == XTypes::TK_NONE) {\n"
```

Comment:
Move lock within the `if`?

Reply:
One thread could return a partially constructed type identifier if the lock is within the `if`.  If anything, the lock should be first.  I'll fix the other instance as well.

Reply:
Sorry, I misread it.